### PR TITLE
[JENKINS-43733] Pick up the latest release of version-number library

### PR DESCRIPTION
**Pull request by recena**
# Description

See [JENKINS-43733](https://issues.jenkins-ci.org/browse/JENKINS-43733).

Details: The main motivation is to keep this library updated on Jenkins core.

[Changes](https://github.com/jenkinsci/lib-version-number/compare/version-number-1.3...version-number-1.4) between `version-number-1.3` and `version-number-1.4`: https://github.com/jenkinsci/lib-version-number/pull/3

### Changelog entries

Proposed changelog entries:

* N/A

### Submitter checklist

- [x] JIRA issue is well described
- [x] Link to JIRA ticket in description, if appropriate

### Desired reviewers

@jenkinsci/code-reviewers
@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
No need in the placeholder JIRA if it is an improvement without backporting need. 🐝 anyway

**Review by jglick — CHANGES_REQUESTED** [review]
You should fix `OldDataMonitor.VersionRange.isOld` now that there is a replacement, as mentioned in https://github.com/jenkinsci/lib-version-number/pull/3.

**Comment by recena** [comment]
@jglick Addressed.

**Comment by recena** [comment]
The PR builder is failing due to something unrelated with these changes:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project test: ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was cmd.exe /X /C "C:\Packages\Plugins\Microsoft.Compute.CustomScriptExtension\1.8\Downloads\0\tools\hudson.model.JDK\jdk8\jre\bin\java -Dfile.encoding=UTF-8 -Xmx1g -jar C:\Packages\Plugins\Microsoft.Compute.CustomScriptExtension\1.8\Downloads\0\workspace\Core_jenkins_PR-2851-7NNP5F22RZD2R5X4EAH2CSUY2YJI56S3FC4YDZ2LVB3IL262SFSA\test\target\surefire\surefirebooter156011780002022759.jar C:\Packages\Plugins\Microsoft.Compute.CustomScriptExtension\1.8\Downloads\0\workspace\Core_jenkins_PR-2851-7NNP5F22RZD2R5X4EAH2CSUY2YJI56S3FC4YDZ2LVB3IL262SFSA\test\target\surefire\surefire3610675378673748306tmp C:\Packages\Plugins\Microsoft.Compute.CustomScriptExtension\1.8\Downloads\0\workspace\Core_jenkins_PR-2851-7NNP5F22RZD2R5X4EAH2CSUY2YJI56S3FC4YDZ2LVB3IL262SFSA\test\target\surefire\surefire_58895678356336142184tmp"
[ERROR] -> [Help 1]
```

The tests were:

```
Tests run: 7877, Failures: 0, Errors: 0, Skipped: 5, Flakes: 1
```

**Comment by oleg-nenashev** [comment]
may be related if version-number bundles JNI stuff somehow, but I doubt about that

**Comment by recena** [comment]
@oleg-nenashev I don't think so. I linked the changes between `1.3` and `1.4`.

**Review by daniel-beck — COMMENTED** [review]
Would prefer if this were held off a bit until https://github.com/jenkinsci/lib-version-number/pull/4 has been considered.

**Comment by recena** [comment]
@daniel-beck What do you mean with _has been considered_? Tested? Verified? Checked?

**Comment by oleg-nenashev** [comment]
@recena I suppose @daniel-beck wants to avoid API deprecation by just updating to `1.5` with his proposal. Since this PR has missed 2.56, it seems reasonable to me to do that in 2.58 on next Sunday

**Comment by recena** [comment]
@oleg-nenashev ah ok. Thanks for your explanation.

**Comment by daniel-beck** [comment]
@recena AFAIUI, the change to `OldDataMonitor` would be unnecessary if my PR is considered useful and merged into `version-number`. If it's not, we can just go ahead here.

File: test/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 !*.zip
+/0/
```
Comment (jglick) [inline test/.gitignore]: Huh? Please revert.

**Review by jglick — CHANGES_REQUESTED** [review]

File: test/.gitignore (hunk 1 above)
Comment (recena) [inline test/.gitignore]: @jglick When you run the tests this folder is created. I though it would be interesting to ignore it.

Comment (jglick) [inline test/.gitignore]: Created by what? I have never seen such a file. I think something may be broken on your machine.

Comment (daniel-beck) [inline test/.gitignore]: I've seen these directories as well.

Comment (jglick) [inline test/.gitignore]: Containing what?

Comment (recena) [inline test/.gitignore]: `mvn test -Dtest=OldDataMonitorTest`

output:

```
ip-192-168-1-111:test recena$ cd 0/
ip-192-168-1-111:0 recena$ ls -la
total 8
drwxr-xr-x  3 recena  staff  102 26 abr 18:31 .
drwxr-xr-x  8 recena  staff  272 26 abr 18:31 ..
-rw-r--r--  1 recena  staff  197 26 abr 18:31 build.xml
ip-192-168-1-111:0 recena$ 
```

This is being generated and hence, it should be ignored.

Comment (olivergondza) [inline test/.gitignore]: I would say it should be generated somewhere where it gets cleaned ...

Comment (jglick) [inline test/.gitignore]: Nope, fixing in #2860, so please revert.

Comment (recena) [inline test/.gitignore]: @jglick thanks.

**Review by jglick — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
lgtm

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
Will merge tomorrow if there is no feedback

**Pull request merged**